mutation_partition_v2.cc

```diff
@@ -0,0 +1,1222 @@
+                // Increment is needed, we don't want to set attributes on the lower bound of the interval.
+                prev_lb_i = lb_i;
+                ++lb_i;
+            } else {
+                lb_i = _rows.begin();
+            }
+
+            while (lb_i != i) {
+                bool compaction_worthwhile = src_e.range_tombstone() > lb_i->range_tombstone();
+
+                // This works for both evictable and non-evictable snapshots.
+                // For evictable snapshots we could replace the tombstone with newer, but due to
+                // the "information monotonicity" rule, adding tombstone works too.
+                lb_i->set_range_tombstone(lb_i->range_tombstone() + src_e.range_tombstone());
+                lb_i->set_continuous(true);
+
+                if (prev_compacted && prev_lb_i) {
+                    maybe_drop(s, tracker, prev_lb_i, app_stats);
+                }
+
+                prev_compacted = false;
+                if (lb_i->dummy()) {
+                    prev_compacted = true;
+                } else if (compaction_worthwhile) {
+                    compact(*lb_i);
+                    prev_compacted = true;
+                }
+
+                if (need_preempt()) {
+                    auto s1 = alloc_strategy_unique_ptr<rows_entry>(
+                            current_allocator().construct<rows_entry>(s,
+                                 position_in_partition::after_key(s, lb_i->position()), is_dummy::yes, is_continuous::no));
+                    alloc_strategy_unique_ptr<rows_entry> s2;
+                    if (lb_i->position().is_clustering_row()) {
+                        s2 = alloc_strategy_unique_ptr<rows_entry>(
+                                current_allocator().construct<rows_entry>(s, s1->position(), is_dummy::yes, is_continuous::yes));
+                        auto lb_i_next = std::next(lb_i);
+                        if (lb_i_next != _rows.end() && lb_i_next->continuous()) {
+                            s2->set_range_tombstone(lb_i_next->range_tombstone() + src_e.range_tombstone());
+                        } else {
+                            s2->set_range_tombstone(src_e.range_tombstone());
+                        }
+                    }
+                    p_sentinel = std::move(s1);
+                    this_sentinel = std::move(s2);
+                    mplog.trace("preempted, res={}", res);
+                    return stop_iteration::no;
+                }
+
+                prev_lb_i = lb_i;
+                ++lb_i;
+            }
+        }
+
+        auto next_p_i = std::next(p_i);
+
+        // next_interval_loaded is true iff there are attributes on next_p_i which apply
+        // to the interval (p_i->position(), next_p_i->position), and we
+        // have to prepare a sentinel when removing p_i from p in case merging
+        // needs to stop before next_p_i is moved.
+        // FIXME: This will be very inefficient for non-evictable snapshots.
```

Comment:
Is this FIXME still relevant after `mutation_partition_v2: Avoid full scan when applying mutation to non-evictable`, or is it a leftover?